jersey-servers/src/test/java/com/palantir/remoting3/servers/jersey/DeprecationWarningFilterTest.java

```diff
@@ -0,0 +1,117 @@
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+
+public final class DeprecationWarningFilterTest {
+
+    @ClassRule
+    public static final DropwizardAppRule<Configuration> APP =
+            new DropwizardAppRule<>(TestServer.class, "src/test/resources/test-server.yml");
+
+    private WebTarget target;
+
+    @Before
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+        String endpointUri = "http://localhost:" + APP.getLocalPort();
+        JerseyClientBuilder builder = new JerseyClientBuilder();
+        Client client = builder.build();
+        target = client.target(endpointUri);
+    }
+
+    @Test
+    public void addsHeaderForDeprecatedEndpoints() {
+        Response response = target.path("/deprecated").request().get();
+        assertThat(response.getHeaderString(HttpHeaders.WARNING))
+                .isEqualTo("299 - \"Service API endpoint is deprecated: /deprecated\"");
+
+        response = target.path("/deprecated-in-resource").request().get();
+        assertThat(response.getHeaderString(HttpHeaders.WARNING))
+                .isEqualTo("299 - \"Service API endpoint is deprecated: /deprecated-in-resource\"");
+    }
+
+    @Test
+    public void doesNotAddHeaderForNonDeprecatedEndpoints() {
+        Response response = target.path("/not-deprecated").request().get();
+        assertThat(response.getHeaderString(HttpHeaders.WARNING)).isNullOrEmpty();
+    }
+
+    public static class TestServer extends Application<Configuration> {
+        @Override
+        public final void run(Configuration config, final Environment env) throws Exception {
+            env.jersey().register(DeprecationWarningFilter.INSTANCE);
+            env.jersey().register(new TestResource());
+        }
+    }
+
+    public static final class TestResource implements TestService {
+        @Override
+        public void deprecated() {}
+
+        /**
+         * @deprecated foo
+         */
+        @Override
+        @Deprecated
+        public void deprecatedInResource() {}
+
+        @Override
+        public void notDeprecated() {}
+    }
+
+    @Path("/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public interface TestService {
+        /**
+         * @deprecated foo
+         */
+        @GET
+        @Path("/deprecated")
+        @Deprecated
+        void deprecated();
+
+        @GET
+        @Path("/deprecated-in-resource")
```

Comment:
include templates to make sure you don't send back the unsafe params

Reply:
done